### PR TITLE
fix(DB/creature_template): Port Immunities for Ogrila quest npcs from Mangos.

### PR DESCRIPTION
**Pull request by avarishd**
## Changes Proposed:
- Port immunities from mangos

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/5167
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/15354

## SOURCE:

Mangos
## Tests Performed:

- not tested

## How to Test the Changes:

.go c id 22199,20600,20216,22910,22911
See immunities

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1681810720948358500.sql [hunk 1]
```diff
@@ -0,0 +1,7 @@
+-- Bosses on the Ogrila quest chain
+-- Slaag + no dodge
+UPDATE `creature_template` SET `mechanic_immune_mask` = `mechanic_immune_mask`|545470038, `flags_extra` = `flags_extra`|8388608 WHERE `entry` = 22199;
+-- Maggoc, Grulloc, Skulloc Soulgrinder
```
Comment (Gultask) [inline data/sql/updates/pending_db_world/rev_1681810720948358500.sql:4]: I'm not too sure about the dodge. Only 2 creatures in AC have it. TC doesn't have it at all

Comment (avarishd) [reply data/sql/updates/pending_db_world/rev_1681810720948358500.sql:4]: Yeah, neither am I. Probably best not to add.

**Review by Gultask — APPROVED** [review]

**Comment by T-Lorentzen** [comment]
You made Grulloc immune to Sablemane's Sleeping Powder with this update.

**Comment by avarishd** [comment]
This works, but It may be that SPELL_ATTR0_NO_IMMUNITIES[ (Unaffected by invulnerability)](https://www.wowhead.com/wotlk/spell=38510) should bypass mechanic_immune_mask?
```c++
    // Sablemane's Sleeping Powder
    ApplySpellFix({ 38510 }, [](SpellInfo* spellInfo)
    {
        spellInfo->Mechanic = 0;
    });
```

**Comment by Gultask** [comment]
I can confirm that Grulloc is immune to Wyvern Sting (Sleep)

Also @T-Lorentzen you can remove the mechanic for now by applying into your local:
```
UPDATE `creature_template` SET `mechanic_immune_mask` = `mechanic_immune_mask`&~512 WHERE (`entry` = 20216);
```
then reloading the template and killing Grulloc

If you play on ChromieCraft then I already did that for now

**Comment by Kitzunu** [comment]
Avarshids comment does sound reasonable

**Pull request merged**